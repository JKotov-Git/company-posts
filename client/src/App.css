* {
  margin: 0;
}

.App {
  width: 100%;
  height: 100vh;
  padding: 0;
  box-sizing: border-box;
}
