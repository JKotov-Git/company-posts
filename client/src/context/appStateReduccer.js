export const initialState = {
  user: null,
};

export const reducer = (state, action) => {
  switch (action.type) {
    case "SIGNIN_USER":
      return { ...state, user: action.userObject };

    default:
      return { ...state };
  }
};
