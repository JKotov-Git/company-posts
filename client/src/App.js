import "./App.css";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import HomePage from "../src/pages/HomePage/HomePage";
import LoginPage from "../src/pages/LoginPage/LoginPage";
import CreateAccount from "../src/pages/CreateAccount/CreateAccount";
import CreatePost from "../src/pages/CreatePost/CreatePost";
import ReadPost from "../src/pages/ReadPost/ReadPost";
import Header from "../src/components/header/Header";
import Footer from "../src/components/footer/Footer";

function App() {
  return (
    <Router>
      <div className="App">
        <Switch>
          <Route path="/readPost">
          <Header />
            <ReadPost />
            <Footer />
          </Route>
          <Route path="/createPost">
            <Header />
            <CreatePost />
            <Footer />
          </Route>
          <Route path="/createAccount">
            <CreateAccount />
          </Route>
          <Route path="/login">
            <LoginPage />
          </Route>
          <Route path="/">
            <Header />
            <HomePage />
            <Footer />
          </Route>
        </Switch>
        
      </div>
    </Router>
  );
}

export default App;
