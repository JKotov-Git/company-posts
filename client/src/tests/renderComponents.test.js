import React from "react";
import App from "../App";
import ReactDOM from "react-dom";
import { configure, shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import { initialState, reducer } from "../context/appStateReduccer";
import StateProvider from "../context/StateProvider";

import { cleanup } from "@testing-library/react";
describe("render react component ", () => {
  //
});

afterEach(cleanup);
