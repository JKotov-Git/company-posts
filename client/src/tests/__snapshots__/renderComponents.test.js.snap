// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="App"
>
  <header
    className="header"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root header-home-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
        />
      </svg>
    </a>
    <div
      className="header-title"
    >
      <h2
        data-testid="headerTitle"
      >
        Company Posts
      </h2>
    </div>
    <div
      className="header-nav"
    >
      <a
        className="header-link"
        href="/login"
        onClick={[Function]}
      >
        <div
          className="header-option"
        >
          <h5>
            Hello, 
          </h5>
          <h5>
            Sign In
          </h5>
        </div>
      </a>
    </div>
  </header>
  <div
    className="home"
  >
    <div
      className="navbar"
    >
      <ul
        className="navbar-list"
      >
        <li
          className="navbar-list-item"
          data-testid="navbarPosts"
        >
          Posts
        </li>
        <a
          className="navbar-list-item-link"
          href="/createPost"
          onClick={[Function]}
        >
          <li
            className="navbar-list-item"
            data-testid="navbaNewPost"
          >
            Add new post
          </li>
        </a>
      </ul>
      <div
        className="navbar-search"
      >
        <input
          className="navbar-search-box"
          data-testid="searchInputField"
          onChange={[Function]}
          placeholder="Search..."
          type="text"
        />
        <div
          className="navbar-search-options"
        >
          <h5
            className="navbar-search-option-title"
          >
            Search by:
          </h5>
          <div
            className="navbar-search-option-by-post"
          >
            <input
              checked={true}
              className="search-option"
              data-testid="searchByPostTitleCheckbox"
              onChange={[Function]}
              type="checkbox"
            />
            <label>
              post name
            </label>
          </div>
          <div
            className="navbar-search-option-by-user"
          >
            <input
              checked={false}
              className="search-option"
              data-testid="searchByUsernameCheckbox"
              onChange={[Function]}
              type="checkbox"
            />
            <label>
              user name
            </label>
          </div>
        </div>
        <button
          data-testid="btnSortByTitle"
          onClick={[Function]}
          type="button"
        >
          Sort By post
        </button>
      </div>
    </div>
    <div
      className="posts-list-container"
    >
      <h4
        className="no-posts-yet-message"
      >
        No Posts Yet.
      </h4>
    </div>
    <div
      className="post-list-paginate-container"
    >
      <ul>
        <li
          className="previous disabled"
        >
          <a
            aria-disabled="true"
            aria-label="Previous page"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
          >
            Previous
          </a>
        </li>
        <li
          className="next"
        >
          <a
            aria-disabled="false"
            aria-label="Next page"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
          >
            Next
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="footer-container"
  >
    <div
      className="footer-company"
    >
      <h2>
        Company Posts
      </h2>
    </div>
    <div
      className="footer-copyright"
    >
      <h4>
        Copyright
      </h4>
    </div>
    <div
      className="footer-links"
    >
      <ul>
        <li>
          Posts
        </li>
        <li>
          Users
        </li>
        <li>
          Sign In
        </li>
      </ul>
    </div>
  </div>
</div>
`;
